ci-operator/step-registry/openstack-k8s-operators/deploy/openstack-k8s-operators-deploy-commands.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+set -ex
+
+OPENSTACK_OPERATOR="openstack-operator"
+
+# We don't want to use OpenShift-CI build cluster namespace
+unset NAMESPACE
+
+# PR SHA
+PR_SHA=$(echo ${JOB_SPEC} | jq -r '.refs.pulls[0].sha')
+
+export IMAGE_TAG_BASE=${REGISTRY}/${ORGANIZATION}/${OPENSTACK_OPERATOR}
+export OPENSTACK_OPERATOR_INDEX=${IMAGE_TAG_BASE}-index:${PR_SHA}
+
+if [ ! -d "${HOME}/install_yamls" ]; then
+  cd ${HOME}
+  git clone https://github.com/openstack-k8s-operators/install_yamls.git
+fi
+
+cd ${HOME}/install_yamls
+# Create/enable openstack namespace
+make namespace
+# Creates storage
+# Sometimes it fails to find container-00 inside debug pod
+# TODO: fix issue in install_yamls
+n=0
+retries=3
+while true; do
+  make crc_storage && break
+  n=$((n+1))
+  if (( n >= retries )); then
+    echo "Failed to run 'make crc_storage' target. Aborting"
+    exit 1
+  fi
+  sleep 10
+done
+
+
+# Deploy openstack operator
+make openstack OPENSTACK_IMG=${OPENSTACK_OPERATOR_INDEX}
+# Wait before start checking all deployment status
+# Not expecting to fail here, only in next deployment checks
+n=0
+retries=30
+until [ "$n" -ge "$retries" ]; do
+  oc get deployment openstack-operator-controller-manager && break
+    n=$((n+1))
+    sleep 10
+done
+
+# Check if all deployments are available
+INSTALLED_CSV=$(oc get subscription openstack-operator -o jsonpath='{.status.installedCSV}')
+oc get csv ${INSTALLED_CSV} -o jsonpath='{.spec.install.spec.deployments[*].name}' | \
+timeout ${TIMEOUT_OPERATORS_AVAILABLE} xargs -I {} -d ' ' \
+sh -c 'oc wait --for=condition=Available deployment {} --timeout=-1s'
+
+# Deploy openstack services
+make openstack_deploy
+sleep 60
+# Debug
+oc get OpenStackControlPlane openstack -o json
+# Waiting for all services to be ready
+oc get OpenStackControlPlane openstack -o jsonpath='{.status.conditions[*].type}' | \
+timeout ${TIMEOUT_SERVICES_READY} xargs -I {} -d ' ' \
+sh -c 'echo testing condition={}; oc wait openstackcontrolplane.core.openstack.org/openstack --for=condition={} --timeout=-1s'
+
+# DEBUG - Getting resources information
+oc get all
+
+# Create clouds.yaml file to be used in further tests.
+mkdir -p ~/.config/openstack
```

Comment:
@viroel I'm wondering if we can remove L72-L85 once [1] merges, and reduce this part to:

```
 oc rsh openstackclient
 openstack token issue
 ```
 
 Using a `client` pod is easier and reduce the maintenance over this part in the future. Not sure we want to `depends-on` [1] but it's something to think about.
 
 [1] https://github.com/openstack-k8s-operators/openstack-operator/pull/117

Reply:
Sure, we can, once openstackclient is available. Prow doesn't support depends-on, so we can't do speculative tests here.